e2e/epinio.e2e.spec.ts

```diff
@@ -0,0 +1,238 @@
+  });
+  test('should install epinio-installer application', async() => {
+    const loadBalancerIpAddr = await loadBalancerIp();
+    const epinioInstall = await helm('install', 'epinio-installer', 'epinio/epinio-installer',
+      '--set', 'skipTraefik=true', '--set', `domain=${ loadBalancerIpAddr }.omg.howdoi.website`,
+      '--wait', '--timeout=25m');
+
+    expect(epinioInstall).toContain('STATUS: deployed');
+  });
+  test('should update epinio config certs file', async() => {
+    const epinioConfigUpdate = await epinio('config', 'update');
+
+    expect(epinioConfigUpdate).toContain('Ok');
+  });
+  test('should push a sample app through epinio cli', async() => {
+    const epinioPush = await epinio('push', '--name', 'sample', '--path', path.join(__dirname, 'assets', 'sample-app'));
+
+    expect(epinioPush).toContain('App is online.');
+  });
+  test('should verify deployed sample application is reachable', async() => {
+    const loadBalancerIpAddr = await loadBalancerIp();
+    const urlAddr = `https://sample.${ loadBalancerIpAddr }.omg.howdoi.website`;
+    // Trick to avoid error 60 (SSL Cert error), passing "--insecure" parameter
+    const sampleApp = await curl('--fail', '--insecure', urlAddr);
+
+    expect(sampleApp).toContain('PHP Version');
+  });
+});
+
+/**
+ * Helper to identify the Load Balancer IP Address.
+ * It will return the traefik IP address, required by epinio install.
+ */
+export async function loadBalancerIp() {
+  const serviceInfo = await kubectl('describe', 'service', 'traefik', '--namespace', 'kube-system');
+
+  const serviceFiltered = serviceInfo.split('\n').toString();
+  const ipAddrRegex = /(LoadBalancer Ingress:)\s+(((?:[0-9]{1,3}\.){3}[0-9]{1,3}))/;
+  const regex = new RegExp(`${ ipAddrRegex.source }`);
+  const ipAddressLb = regex.exec(serviceFiltered);
+
+  // checking if it will be undefined, null, 0 or empty
+  if (typeof ipAddressLb !== 'undefined' && ipAddressLb) {
+    return ipAddressLb[2];
+  } else {
+    console.log('Cannot find load balancer IP address.');
+  }
+}
+
+export async function installEpinioCli() {
+  // Should detect which OS we're running this spec in and download the correct binary dynamically
+  const platform = detectPlatform();
+
+  // Download epinio binary based on platform type
+  await downloadEpinioBinary(platform as string);
+}
+
+/**
+ * Download epinio binary based on the platform type and save the binary
+ * into a temporary folder.
+ */
+export async function downloadEpinioBinary( platformType: string) {
+  // Setting up epinio binaries names per platform
+  const epinioWin = 'epinio-windows-amd64.exe';
+  const epinioLinux = 'epinio-linux-x86_64';
+  const epinioDarwin = 'epinio-darwin-x86_64';
+  const epinioDarwinArm = 'epinio-darwin-arm64';
+  const epinioWorkingVersion = 'v0.5.0';
+
+  // Create a temp folder for epinio binary
+  const epinioTempFolder = path.join(os.homedir(), 'epinio-tmp');
```

Comment:
Should probably use [`os.tmpdir()`](https://nodejs.org/docs/latest-v16.x/api/os.html#ostmpdir).